viadot/tasks/customer_gauge.py

```diff
@@ -52,13 +66,238 @@ def __init__(
+
+        Returns:
+            Dict[str, Any]: The JSON response with modified nested dictionaries
+        within the specified field.
+        """
+        d={}
+        for i, dictionary in enumerate(json_response[field], start=1):
+            for key, value in dictionary.items():
+                d[f'{i}_{key}'] = value
+        if d:
+            json_response[field] = d
+
+        return json_response
+    
+    def column_unpacker(
+        self, 
+        json_list: List[Dict[str, Any]] = None,
+        method1_cols: List[str] = None,
+        method2_cols: List[str] = None,
+        ) -> List[Dict[str, Any]]:
+
+        """
+        Unpack and modify specific columns in a list of dictionaries using two methods, chosen by the user. 
+        If user wants to use field_reference_unpacker, he needs to provide list of fields in `method1_cols` 
+        argument, if user wants to use nested_dict_transformer - uses 'method2_cols' argument.
+
+        Args:
+            json_list (List[Dict[str, Any]): A list of dictionaries containing the data.
+            method1_cols (List[str]): Columns to unpack and modify using `_field_reference_unpacker`. 
+                Defaults to None.
+            method2_cols (List[str]): Columns to unpack and modify using `_nested_dict_transformer`. 
+                Defaults to None.
+
+        Raises:
+            ValueError: _description_
+
+        Returns:
+            List[Dict[str, Any]]: The updated list of dictionaries after column unpacking and modification.
+        """
+
+        if json_list is None:
+            raise ValueError("Input 'json_list' is required.")
+
+        def unpack_columns(columns, unpack_function):
+            json_list_clean = json_list.copy()
+            for field in columns:
+                if field in json_list_clean[0]:
+                    logger.info(f"Unpacking column '{field}' with {unpack_function.__name__} method...")
+                    try:
+                        json_list_clean = list(map(lambda x: unpack_function(x, field), json_list_clean))
+                        logger.info(f"All elements in '{field}' are unpacked successfully.")
+                    except:
+                        logger.info(f"No transformation were made in '{field}'," 
+                        "because didn't contain list of key-value data.")
+                else:
+                    logger.info(f"Column '{field}' not found.")
+            return json_list_clean
+
+        if method1_cols is not None:
+            json_list = unpack_columns(
+                columns = method1_cols, 
+                unpack_function = self._field_reference_unpacker
+                )
+
+        if method2_cols is not None:
+            json_list = unpack_columns(
+                columns = method2_cols, 
+                unpack_function = self._nested_dict_transformer
+                )
+        
+        return json_list
```

Comment:
Both of them can be executed, because different fields might need to be unpacked in two different ways. But fair point about overwritting, I'll add logger to let user know that he mentioned the same column in both methods.